libs/ui/src/components/activity-section-list/activity-section-list.tsx

```diff
@@ -36,67 +35,110 @@ interface Props {
 
 export const ActivitySectionList: FC<Props> = ({ publicKeyHash, chainId, filterTypeName, tokenAddress = '' }) => {
   const [offsetY, setOffsetY] = useState(0);
-  const { activity: allActivity, fetch, isLoading } = useAllActivity(publicKeyHash, getDebankId(chainId), tokenAddress);
+
+  const {
+    activity: allActivity,
+    fetch,
+    isLoading
+  } = useAllActivity(publicKeyHash, getDebankId(chainId), filterTypeName, tokenAddress);
+
+  const sectionListRef = useRef<SectionList<ActivityData, SectionListActivityData>>(null);
 
   const activity = useMemo(() => getFilteredActivity(allActivity, filterTypeName), [allActivity, filterTypeName]);
 
+  const activityDataLength = useMemo(() => {
+    let sum = 0;
+
+    if (sum <= DEFAULT_VISIBILITY_VALUE) {
+      for (const { data } of activity) {
+        sum += data.length;
+      }
+    }
+
+    return sum;
+  }, [activity]);
+
   useEffect(() => {
     if (
       filterTypeName !== ActivityFilterEnum.AllActivity &&
       allActivity.length > 0 &&
-      activity.length === 0 &&
+      activityDataLength <= DEFAULT_VISIBILITY_VALUE &&
       numberOfAttempts <= MAX_ACTIVITY_ATTEMPTS
     ) {
       const lastDate = allActivity.slice(-1)[0].data.slice(-1)[0].timestamp;
 
       fetch(lastDate);
       numberOfAttempts += 1;
     }
-  }, [filterTypeName, allActivity, activity]);
+  }, [filterTypeName, allActivity, activityDataLength]);
 
-  const handleFetchData = () => {
+  const handleFetchData = useCallback(() => {
     if (offsetY === 0) {
       fetch(0);
       numberOfAttempts = 0;
     }
-  };
+  }, [offsetY]);
 
-  useTimerEffect(handleFetchData, DATA_UPDATE_TIME, [publicKeyHash, chainId, offsetY]);
+  useTimerEffect(handleFetchData, DATA_UPDATE_TIME, [publicKeyHash, chainId, offsetY, filterTypeName]);
 
-  const handleScrollWeb = debounce((event: NativeSyntheticEvent<NativeScrollEvent>) => {
-    setOffsetY(event.nativeEvent.contentOffset.y);
-  }, DEBOUNCE_TIME);
+  const debounceContentOffset = debounce(
+    (event: NativeSyntheticEvent<NativeScrollEvent>) => setOffsetY(event.nativeEvent.contentOffset.y),
+    DEBOUNCE_TIME
+  );
 
-  const handleScrollMobile = (event: NativeSyntheticEvent<NativeScrollEvent>) => {
-    event.persist();
-    debounce(() => setOffsetY(event.nativeEvent.contentOffset.y), DEBOUNCE_TIME);
-  };
+  const handleScroll = useCallback((event: NativeSyntheticEvent<NativeScrollEvent>) => {
+    if (isMobile) {
+      event.persist();
+    }
+    debounceContentOffset(event);
+  }, []);
 
-  const handleEndReached = () => {
-    fetch();
-  };
+  const handleEndReached = async ({ distanceFromEnd }: { distanceFromEnd: number }) => {
+    if (distanceFromEnd >= 0) {
+      fetch();
+    }
 
-  const renderItem: SectionListRenderItem<ActivityData, SectionListActivityData> = useCallback(
-    ({ item }) => <ActivityItem transaction={item} address={publicKeyHash} chainName={getDebankId(chainId)} />,
-    [publicKeyHash, chainId]
-  );
+    await sleep(100);
```

Comment:
we need this to avoid new scroll before new activity items is rendered? 
what if it took more than 100 ms ?

Reply:
We need the sleep function because we need to wait for the Loading state and render Loader, without the sleep we just get the bottom of the list Loader is not shown and we can't scroll